test/mongodb.test.js

```diff
@@ -717,6 +718,51 @@ describe('mongodb connector', function() {
     });
   });
 
+  it('should not return data for nested `$where` in where', function(done) {
+    Post.create({title: 'Post1', content: 'Post1 content'}, (err, p1) => {
+      Post.create({title: 'Post2', content: 'Post2 content'}, (err2, p2) => {
+        Post.create({title: 'Post3', content: 'Post3 data'}, (err3, p3) => {
+          Post.find({where: {$where: 'function() {return this.content.contains("content")}'}}, (err, p) => {
+            should.not.exist(err);
+            p.length.should.be.equal(3);
+            done();
+          });
+        });
+      });
+    });
+  });
+
+  it('should allow $where in where with options.disableSanitization', function(done) {
+    Post.create({title: 'Post1', content: 'Post1 content'}, (err, p1) => {
+      Post.create({title: 'Post2', content: 'Post2 content'}, (err2, p2) => {
+        Post.create({title: 'Post3', content: 'Post3 data'}, (err3, p3) => {
+          Post.find(
+            {where: {$where: 'function() {return this.content.contains("content")}'}},
+            {disableSanitization: true},
+            (err, p) => {
+              should.not.exist(err);
+              p.length.should.be.equal(2);
+              done();
+            });
+        });
+      });
+    });
+  });
+
+  it('does not execute a nested `$where`', function(done) {
+    Post.create({title: 'Post1', content: 'Post1 content'}, (err, p1) => {
+      Post.create({title: 'Post2', content: 'Post2 content'}, (err2, p2) => {
+        Post.create({title: 'Post3', content: 'Post3 data'}, (err3, p3) => {
+          Post.find({where: {content: {$where: 'function() {return this.content.contains("content")}'}}}, (err, p) => {
+            should.not.exist(err);
+            p.length.should.be.equal(0);
```

Comment:
Let me know if I'm understanding this correctly: $where clause here is not executed, meaning the where filter effectively looks like this: `{where: {content: ''}}`, so `p` in this case is just an empty array.

Reply:
Well ... it tried to match `content` to the object `{$where: '...'}`